Scale Collision Pick to match parent

sabrina-shanman opened the pull request:
This is a last-minute change to the Collision Pick API which makes collision picks scale to match their parent, for better use in the safe teleport script.

Old behavior:
- Collision Picks parented to avatars, entities, and overlays do not scale to match their parent

New Behavior:
- Collision Picks parented to avatars, entities, and overlays rescale as their parent changes size

## TEST PLAN
- Perform the manual test at https://github.com/sabrina-shanman/hifi_tests/blob/safe-teleport-target/tests/engine/interaction/pick/collision/parenting_on_server/story.md

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12699/HighFidelity-Beta-PR13949-84a2512.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12699/HighFidelity-Beta-Interface-PR13949-84a2512.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12699/HighFidelity-Beta-PR13949-84a2512.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12699/HighFidelity-Beta-Interface-PR13949-84a2512.exe

SamGondelman commented on interface/src/raypick/PickScriptingInterface.cpp:
```diff
@@ -375,6 +377,16 @@ std::shared_ptr<TransformNode> PickScriptingInterface::createTransformNode(const
+                if (nestableType == NestableType::Overlay) {
+                    return std::make_shared<OverlayTransformNode>(std::static_pointer_cast<Base3DOverlay>(sharedNestablePointer), parentJointIndex);
+                }
+                if (nestableType == NestableType::Entity) {
```
These can be `else if`s

SamGondelman commented on interface/src/ui/overlays/OverlayTransformNode.cpp:
```diff
@@ -0,0 +1,30 @@
+    bool success;
+    Transform jointWorldTransform = overlay->getTransform(_jointIndex, success);
+    if (!success) {
+        jointWorldTransform = Transform();
```
This can just be `return Transform()` (I don't think it makes sense to set the scale if this fails)

SamGondelman commented on libraries/avatars-renderer/src/avatars-renderer/AvatarTransformNode.cpp:
```diff
@@ -0,0 +1,30 @@
+    bool success;
+    Transform jointWorldTransform = avatar->getTransform(_jointIndex, success);
+    if (!success) {
+        jointWorldTransform = Transform();
```
`return Transform()`

SamGondelman commented on libraries/entities/src/EntityTransformNode.cpp:
```diff
@@ -0,0 +1,30 @@
+    bool success;
+    Transform jointWorldTransform = entity->getTransform(_jointIndex, success);
+    if (!success) {
+        jointWorldTransform = Transform();
```
`return Transform()`

sabrina-shanman commented:
I also addressed a compiler error and corrected an oversight where the threshold would not scale with the parent. The threshold now scales with the largest dimension of the parent. I should probably change the jsdocs to reflect the changes.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12716/HighFidelity-Beta-PR13949-7839ba4.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12719/HighFidelity-Beta-PR13949-714cfc5.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12719/HighFidelity-Beta-Interface-PR13949-714cfc5.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12719/HighFidelity-Beta-PR13949-714cfc5.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12719/HighFidelity-Beta-Interface-PR13949-714cfc5.exe

sabrina-shanman commented:
Luis' requested change has been added.

Collision picks still scale with their parent if their parent is a Spatially Nestable, however the dimensions of the collision pick are now initially defined in world space, and then scale proportionally with the parent after creation.

sabrina-shanman commented on interface/src/raypick/PickScriptingInterface.cpp:
```diff
@@ -383,6 +383,10 @@ std::shared_ptr<TransformNode> PickScriptingInterface::createTransformNode(const
+            glm::vec3 baseScale = glm::vec3(1);
+            if (propMap["baseScale"].isValid()) {
+                baseScale = vec3FromVariant(propMap["baseScale"]);
+            }
```
Whoops

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12747/HighFidelity-Beta-PR13949-23a6abb.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12747/HighFidelity-Beta-Interface-PR13949-23a6abb.exe

sabrina-shanman commented:
Shoot, bad timing, I just uploaded a change to address a previous failure in the Linux build.

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12754/HighFidelity-Beta-PR13949-52bc09c.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12754/HighFidelity-Beta-Interface-PR13949-52bc09c.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12764/HighFidelity-Beta-PR13949-8d4c47c.apk

sabrina-shanman commented:
I made another change to comply with the non-Windows compilers.

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12764/HighFidelity-Beta-PR13949-8d4c47c.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12764/HighFidelity-Beta-Interface-PR13949-8d4c47c.dmg

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12764/HighFidelity-Beta-Interface-PR13949-8d4c47c.exe

qaAdamSmith82 commented:
Currently Testing

qaAdamSmith82 commented:
PASS @conklin94122
